lib/max_subarray.py

```diff
@@ -2,11 +2,23 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
-    if nums == None:
+    if not nums:
         return 0
-    if len(nums) == 0:
-        return 0
-    pass
+    if max(nums) < 0: 
+        return max(nums)
+
+   #Implementation of Kadane's Algorithm
+   
+    maxSum = 0
+    currSum = 0 
+
+    for i in range(len(nums)): 
+        currSum = currSum + nums[i]
+        if(currSum > maxSum): 
+            maxSum = currSum
+        if(currSum < 0):
+            currSum = 0
```

Comment:
Just a little simpler.

```suggestion
        currSum = max(currSum + nums[i], nums[i])
        maxSum = max(currSum, maxSum)
```